src/Donations/Components/DonationsForm.tsx

```diff
@@ -187,6 +214,70 @@ function DonationsForm() {
     }
   }
 
+  const handlePlanetCashDonate = async () => {
+    const _onBehalfDonor = {
+      firstname: onBehalfDonor.firstName,
+      lastname: onBehalfDonor.lastName,
+      email: onBehalfDonor.email,
+    };
+
+    const _gift = {
+      ...giftDetails,
+      message: giftDetails.giftMessage,
+    };
+
+    delete _gift.giftMessage;
+
+    // create Donation data
+    const donationData = {
+      purpose: projectDetails!.purpose,
+      project: projectDetails!.id,
+      units: quantity,
+      prePaid: true,
+      onBehalf: onBehalf,
+      ...(onBehalf && { donor: _onBehalfDonor }),
+      ...(isGift && { gift: _gift }),
+    };
+
+    const cleanedDonationData = cleanObject(donationData);
+
+    const token = await getAccessTokenSilently();
+
+    // @method    POST
+    // @endpoint  /app/donation
+
+    try {
+      const { data, status } = await apiRequest({
+        url: "/app/donations",
+        method: "POST",
+        setshowErrorCard,
+        data: cleanedDonationData,
+        token,
+        headers: {
+          "IDEMPOTENCY-KEY": uuidv4(),
+        },
+      });
+
+      if (status === 200) {
+        setdonation(data);
+        router.replace({
+          query: { ...router.query, step: THANK_YOU },
+        });
+      }
+    } catch (err) {
+      console.error(err);
+      if (err.status === 400) {
+        setPaymentError(err.data.message);
+      } else if (err.status === 500) {
+        setPaymentError(t("errorStatus500"));
+      } else if (err.status === 503) {
+        setPaymentError(t("errorStatus503"));
```

Comment:
These common errors should also rather be handling directly in `apiRequest` or some common error handler, otherwise we have to implement them for every single API call. But that would demand some higher abstraction level. So for now leave it as it is.